applications/CompressiblePotentialFlowApplication/custom_elements/compressible_potential_flow_element.h

```diff
@@ -83,678 +74,223 @@ class CompressiblePotentialFlowElement : public Element
-
-        //taking only positive part
-        for (unsigned int i = 0; i < NumNodes; i++)
-        {
-            if (distances[i] > 0)
-                data.phis[i] = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
-            else
-                data.phis[i] = GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
-        }
-
-        noalias(velocity) = prod(trans(data.DN_DX), data.phis);
-    }
-
-    void ComputeVelocityLowerWakeElement(array_1d<double,Dim>& velocity)
-    {
-        ElementalData<NumNodes, Dim> data;
-
-        //calculate shape functions
-        GeometryUtils::CalculateGeometryData(GetGeometry(), data.DN_DX, data.N, data.vol);
-
-        array_1d<double, NumNodes> distances;
-        GetWakeDistances(distances);
-
-        //taking only negative part
-        for (unsigned int i = 0; i < NumNodes; i++)
-        {
-            if (distances[i] < 0)
-                data.phis[i] = GetGeometry()[i].FastGetSolutionStepValue(VELOCITY_POTENTIAL);
-            else
-                data.phis[i] = GetGeometry()[i].FastGetSolutionStepValue(AUXILIARY_VELOCITY_POTENTIAL);
-        }
-
-        noalias(velocity) = prod(trans(data.DN_DX), data.phis);
-    }
-
-    void CheckWakeCondition()
-    {
-        array_1d<double, Dim> upper_wake_velocity;
-        ComputeVelocityUpperWakeElement(upper_wake_velocity);
-        const double vupnorm = inner_prod(upper_wake_velocity, upper_wake_velocity);
-
-        array_1d<double, Dim> lower_wake_velocity;
-        ComputeVelocityLowerWakeElement(lower_wake_velocity);
-        const double vlownorm = inner_prod(lower_wake_velocity, lower_wake_velocity);
-
-        if (std::abs(vupnorm - vlownorm) > 0.1)
-            std::cout << "WAKE CONDITION NOT FULFILLED IN ELEMENT # " << this->Id() << std::endl;
-    }
-
-    double ComputePressure(const ProcessInfo& rCurrentProcessInfo)
-    {
-        double pressure = 0.0;
-
-        bool active = true;
-        if ((this)->IsDefined(ACTIVE))
-            active = (this)->Is(ACTIVE);
-
-        if (active && !this->Is(MARKER))
-            pressure = ComputePressureNormalElement(rCurrentProcessInfo);
-        else if (active == true && this->Is(MARKER))
-            pressure = ComputePressureWakeElement(rCurrentProcessInfo);
-
-        return pressure;
-    }
-
-    double ComputePressureNormalElement(const ProcessInfo& rCurrentProcessInfo)
-    {
-        const array_1d<double, 3> vinfinity = rCurrentProcessInfo[VELOCITY_INFINITY];
-        const double vinfinity_norm2 = inner_prod(vinfinity, vinfinity);
-
-        KRATOS_ERROR_IF(vinfinity_norm2 < std::numeric_limits<double>::epsilon())
-            << "Error on element -> " << this->Id() << "\n"
-            << "vinfinity_norm2 must be larger than zero." << std::endl;
-
-        array_1d<double, Dim> v;
-        ComputeVelocityNormalElement(v);
+    void AssignLocalSystemSubdividedElement(
+        MatrixType& rLeftHandSideMatrix,
+        Matrix& lhs_positive,
```

Comment:
Besides using CamelCase, I suggest to think of some more meaningful names for some of the parameters. E.g., `data` is not very descriptive.